content/articles/nesting-navigators-and-routing-with-react-navigation-in-react-native/index.md

```diff
@@ -0,0 +1,435 @@
+            component={NewWorkerScreen}
+          />
+          </>
+        </Stack.Navigator>
+      </NavigationContainer>
+    );
+  }
+}
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    alignItems: 'center',
+    justifyContent: 'center',
+    backgroundColor: '#fff',
+  },
+});
+
+export default App;
+```
+
+While revisiting our `App.js` we'll find that we added `NavigationContainer` and `createStackNavigator` to ease and allow navigation from one screen to another. We also added the two screens: `HomeScreen` and `NewWorkerScreen` between the **<Stack.Navigator>** that we will be navigating through and thus navigator is aware of your two screens.
+
+
+### Fourth Step - Adding Buttons to HomeScreen and NewWorkerScreen
+
+Add buttons to switch between the two screens, if necessary. As long as the navigation object is included in the StackNavigator, many of its useful attributes will be transmitted down to your screen.
+
+Add the following code in the HomeScreen.js to add a button and use the same way to another button in the NewWorkerScreen.js.
+
+```js
+<Button
+  title="Add new workers"
+  onPress={() =>
+    this.props.navigation.navigate('NewWokerScreen')
+  }
+/>
+
+```
+Now, in NewWorkerScreen.js, build a button for YHomescreen:
+
+```js
+<Button
+  title="Home"
+  onPress={() => this.props.navigation.navigate('HomeScreen')}
+/>
+```
+### Fifth Step- Passing Data to Other Screens Using Context
+Create an array of potential workers, including Jeff, Kim, and Cal, and an empty array of present workers. In addition, you'll implement a feature that allows users to add new acquaintances to their existing list of contacts.
+Add possible workers and current workers to the component’s state:
+```js
+
+
+class App extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      possibleWorkers: [
+        'Jeff',
+        'Kim',
+        'Cal',
+      ],
+      currentWorkers: [],
+    }
+  }
+
+  
+}
+
+
+```
+Lets now add a function to move a possible Workers into the list of current Workers:
+```js
+
+
+class App extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      possibleWorkers: [
+        'Jeff',
+        'Kim',
+        'Cal',
+      ],
+      currentWorkers: [],
+    }
+  }
+
+  addWorker = (index) => {
+    const {
+      currentWorkers,
+      possibleWorkers,
+    } = this.state
+
+    // Pull wokers out of possibleWorkers
+    const addedwokers = possibleWorkers.splice(index, 1)
+
+    // And put wokers in currentWorkers
+    currentWorkers.push(addedwokers)
+
+    // Finally, update the app state
+    this.setState({
+      currentWorkers,
+      possibleWorkers,
+    })
+  }
+
+}
+```
+#### Adding WorkersContext to App
+To see your workers on HomeScreen.js, you'll need to add them to WorkersScreen.js first. It's possible to add this feature to your screens with context because this project is built on React.

```

Comment:
WorkersScreen.js? We only have HomeScreen and NewWorkersScreen.js at this point in the article.